### PR TITLE
Updated user

### PR DESCRIPTION
**Pull request by ayancovich**

### CODE REVIEW

File: src/user.controller.ts [hunk 1]
```diff
@@ -37,10 +42,26 @@ export class UserController {
         }
 
         try {
-            const savedUser = await this.repository.save(newUser);
+            const hashuser = {
+                key: (bcrypt.hashSync(Date.toString() + newUser.name, this.PASSWORD_HASH_SIZE)).replace(".", "").replace("\/", "").replace("&", ""),
+                user: newUser,
+                creationDate: new Date(),
+            }
+            var userId = 0;
```
Comment (nestorreloaded) [inline src/user.controller.ts]: Please use let instead of var

File: src/user.controller.ts [hunk 2]
```diff
@@ -37,10 +42,26 @@ export class UserController {
...
+            await getManager().transaction(async transactionalEntityManager => {
+                const savedUser: any = await transactionalEntityManager.getRepository(User).save(newUser);
+                hashuser.user = savedUser;
+                userId = savedUser.id;
+                await transactionalEntityManager.getRepository(HashUser).save(hashuser);
+            });
+            try {
+                new RabbitMq("user_created", JSON.stringify({ "destination": [newUser.email], "user_name": newUser.name, "key": hashuser.key }));
+            } catch (error) {
+                console.log(error);
```
Comment (nestorreloaded) [inline src/user.controller.ts]: Here we should return a response and stop the method execution

File: src/user.controller.ts [hunk 3]
```diff
@@ -117,8 +139,108 @@ export class UserController {
...
+    @ContentType("application/json")
+    @OpenAPI({ summary: 'Enable an user' })
+    @ResponseSchema(Response, {
+        contentType: 'application/json',
+        statusCode: '200',
+    })
+    public async enableUser(@Param("key") key: string): Promise<Response> {
+        try {
+            const hashUserRepo = getManager().getRepository(HashUser);
+            const user_hash: any = await hashUserRepo.createQueryBuilder('hashuser')
```
Comment (nestorreloaded) [inline src/user.controller.ts]: Please move this to a HashRepository

File: src/user.controller.ts [hunk 4]
```diff
@@ -117,8 +139,108 @@ export class UserController {
...
 
+    @Get('/disable/:id')
+    @ContentType("application/json")
+    @OpenAPI({ summary: 'Disable an user' })
+    @ResponseSchema(Response, {
+        contentType: 'application/json',
+        statusCode: '200',
+    })
+    public async disableUser(@Param('id') id: number): Promise<Response> {
+        const user = {
```
Comment (nestorreloaded) [inline src/user.controller.ts:223]: We can use the spread operator here

**Review by nestorreloaded — CHANGES_REQUESTED** [review]
Please review the changes

File: src/user.controller.ts (hunk 2 above)
Comment (ayancovich) [reply src/user.controller.ts]: Done

File: src/user.controller.ts (hunk 1 above)
Comment (ayancovich) [reply src/user.controller.ts]: Done

File: src/app.ts [hunk 5]
```diff
@@ -127,20 +127,22 @@ createConnection(config).then(async connection => {
...
   }).then(config => {
     process.env['BCRYPT_HASH'] = config.properties["users.bcrypt-hash-size"];
+    process.env["HASH_EXPIRE_TIME"] = config.properties["users.hash-expire-time"];
   })
 
   // run express application on port 3000
   app.listen(APP_LISTEN_PORT);
 
   console.log('App is ready');
+  return;
```
Comment (josesolaro) [inline src/app.ts]: ups! this need to be deleted.

**Review by josesolaro — CHANGES_REQUESTED** [review]

File: src/app.ts (hunk 5 above)
Comment (ayancovich) [reply src/app.ts]: Ok

Comment (ayancovich) [reply src/app.ts]: k

File: src/user.controller.ts (hunk 4 above)
Comment (ayancovich) [reply src/user.controller.ts:223]: Can't be posible, we don't recieve the request user in the @Param, we only obtain the ID on the URL.

File: src/user.controller.ts (hunk 3 above)
Comment (ayancovich) [reply src/user.controller.ts]: Done

**Review by nestorreloaded — APPROVED** [review]

**Review by josesolaro — APPROVED** [review]

**Pull request merged**